### DISCUSSION TITLE
Cache layer on top of Postgres

### DISCUSSION THREAD

**Discussion by ihadabs** [supabase/supabase#16758] [Feature Requests]
Can you support query caching in front of Postgres?

Something like Prisma Accelerate:
https://www.prisma.io/docs/data-platform/accelerate/what-is-accelerate

It does not have to be the same...
Just a cache layer in front of Postgres.

This will help serverless apps use the cache layer, so we don't need to use Redis or other services for query caching.

Usage:

## With Supabase Client:
```ts
supabase
    .from('books')
    .select('*')
    .cache({ttl: 30, swr: 60});
```

## With Prisma Client:
1. Have a prisma extension specific for the Supabase pooler (with cache layer)
2. Then we could use the extension like the Accelerate extension

NOTE: I think a lot of people don't need the edge thing, so the cache layer could be as simple as checking the query cache in the pooler before sending it to db

**Comment by arshamg** [comment]
This would be super cool - RIbbon would definitely use this
